lib/galaxy/tool_util/deps/mulled/mulled_build.py

```diff
@@ -531,12 +531,7 @@ def target_str_to_targets(targets_raw):
     def parse_target(target_str):
         if "=" in target_str:
             package_name, version = target_str.split("=", 1)
-            build = None
-            if "=" in version:
-                version, build = version.split("=")
-            elif "--" in version:
-                version, build = version.split("--")
-            target = build_target(package_name, version, build)
+            target = build_target(package_name, version)
```

Comment:
I can't see how that won't break on:
```
combinations/hash.tsv
79:bedtools=2.17.0--0,python=2.7.13	bgruening/busybox-bash:0.1	0
94:picard=2.22.8--0,bwa=0.7.17	bgruening/busybox-bash:0.1	0
95:whatshap=1.0--py37h9a982cc_1,tabix=0.2.6--ha92aebf_0	bgruening/busybox-bash:0.1	0
```
Was there any reason not to use my suggestion ?

Reply:
Indeed we would change image names, but the new image names would be the ones which the container resolvers are looking for (in the current implementation).

Reply:
It won't just change the image name, it will not install the correct build.

Reply:
Now I got it. `requirements_to_mulled_targets` also affects the names that the mulled resolvers expect: https://github.com/galaxyproject/galaxy/blob/a7fb5711431df5995b65330b5c99ac0503e8eec9/lib/galaxy/tool_util/deps/container_resolvers/mulled.py#L820C12-L820C42

I will try to finish this next week.

Sorry for the noise and thanks for double/triple checking.